examples/platform/silabs/SiWx917/SiWxWifiInterface.cpp

```diff
@@ -609,11 +607,13 @@ sl_status_t show_scan_results(sl_wifi_scan_result_t * scan_result)
 
         cur_scan_result.ssid_length = strnlen((char *) scan_result->scan_info[idx].ssid,
                                               std::min<size_t>(sizeof(scan_result->scan_info[idx].ssid), WFX_MAX_SSID_LENGTH));
-        chip::Platform::CopyString(cur_scan_result.ssid, cur_scan_result.ssid_length, (char *) scan_result->scan_info[idx].ssid);
+        // cur_scan_result.ssid is of size WFX_MAX_SSID_LENGTH+1, we are safe with the cur_scan_result.ssid_length calculated above
+        chip::Platform::CopyString(cur_scan_result.ssid, cur_scan_result.ssid_length + 1,
```

Comment:
why is the cast to char * required here? Should this not be const char *? Also same question for line 331 above: why are we casting (and using a C-style cast instead of C++)?

Also wondering about copy-string usage: this copies until the first null terminator, however the syntax on scan_results seem to be length-delimited instead. It seems we would rather want some helpers that use ByteSpan or similar.